third_party/move/move-vm/runtime/src/loader/mod.rs

```diff
@@ -1902,6 +1873,30 @@ impl Module {
                 }
             }
 
+            // validate the correctness of struct handle references.
```

Comment:
This is only useful for the time being, when we still load transitive dependencies, right?

Reply:
That's correct. In other words, we load transitive dependencies so that we can check for this property for now, tho we may choose to relax it in the future.